helios-sample/src/main/kotlin/helios/sample/sample.kt

```diff
@@ -45,22 +45,44 @@ fun main(args: Array<String>) {
         println("Successfully decode the json: $it")
     })
 
-    root.selectExtract<String>("name").modify(companyJson) { chrs ->
-        chrs.toUpperCase()
-    }.let(::println)
+    root.name().string.fix().modify(companyJson, String::toUpperCase).let(::println)
 
-    root.select("address").select("street").selectExtract<String>("name").getOption(companyJson).let(::println)
+    root.address().street().name().string.fix().getOption(companyJson).let(::println)
 
-    val employeeLastNames: Traversal<Json, String> = root.select("employees").every().select("lastName").extract()
+    val employeeLastNames: Traversal<Json, String> = root.employees().every().lastName().string.fix()
 
     employeeLastNames.modify(companyJson, String::capitalize).let {
         employeeLastNames.getAll(it)
     }.let(::println)
 
-    root.select("employees").filterIndex { it == 0 }.select("name").extract<String>()
-            .getAll(companyJson).let(::println)
+    root.employees().filterIndex { it == 0 }.select("name").string.getAll(companyJson).let(::println)
+
+    root.employees().every().filterKeys { it == "name" }.string.getAll(companyJson).let(::println)
+
+}
+
+/**
+ * Generated code for Company
+ */
+fun <F> JsonPathFunctions<F>.name(): JsonPathFunctions<F> = select("name")
```

Comment:
Can we make these extension properties via getter to achieve `address.employees` syntax without explicit function invocations?